packages/core/src/baseclient.ts

```diff
@@ -721,7 +721,7 @@ export abstract class BaseClient<O extends ClientOptions> implements Client<O> {
           throw new SentryError(`${beforeSendLabel} returned \`null\`, will not send event.`, 'log');
         }
 
-        const session = scope && scope.getSession();
+        const session = (scope && scope.getSession()) || getIsolationScope().getSession();
```

Comment:
This makes me a bit uncomfortable tbh, not sure if we for now should simply still use `hub.startSession` in browser/node `startSessionTracking`. But this change seems to be necessary for browser, since `scope.getSession()` returned undefined

Reply:
hmm, is this correct in the new implementation, I wonder? Or should we simply _always_ take the one from the isolation scope? 🤔  because the scope here would generally be the final scope being applied and can even be a temporary scope, like `new Scope()` 🤔

Reply:
> hmm, is this correct in the new implementation, I wonder? 

You mean that we set it onto the isolation scope? I think that should be correct 🤔 

> Or should we simply always take the one from the isolation scope? 🤔 

I guess it would be more correct if we're okay with not supporting people adding their own session to their current scope 🤔 Not sure if that's even a thing...

Reply:
Oh wait, for v7 we probably still need to set the session _also_ on the current scope, right? So that `getCurrentHub().getScope().getSession()` still returns the session

Reply:
This probably makes calling `getIsolationScope` obsolete here

Reply:
hmm I guess so, you're right! :(

Reply:
Reworked the top level APIs to also modify the current scope's session. For now this will work and spare us mocking with event processing